Switch from Moose::Any to Moo

hashbangperl opened the pull request:
Based on PR https://github.com/sartak/path-dispatcher/pull/3
Updated to use Types::Standard and Type::Tiny::Class
Tests passing

Commit:
Switch from Moose::Any to Moo

Based on PR https://github.com/sartak/path-dispatcher/pull/3
Updated to use Types::Standard and Type::Tiny::Class
Tests passing

karenetheridge commented on dist.ini:
```diff
@@ -14,8 +14,10 @@ copyright_holder = Shawn M Moore
-Any::Moose = 0.12
+Moo = 0.12
 Try::Tiny  = 0.02
+Type::Tiny::Class = 1
```
These additions are not necessary, due to `[AutoPrereqs]`. Also, "1" isn't any specific release version of this distribution, is it? If a specific version is required, it should be explicitly stated.

hashbangperl replied:
Sorted

karenetheridge commented on lib/Path/Dispatcher/Match.pm:
```diff
@@ -1,42 +1,44 @@
 has path => (
     is       => 'ro',
-    isa      => 'Path::Dispatcher::Path',
+    isa      => Type::Tiny::Class->new( class => 'Path::Dispatcher::Path'),
```
`class_type` (exported from Type::Utils) is better.

karenetheridge commented on lib/Path/Dispatcher/Path.pm:
```diff
@@ -1,29 +1,31 @@
     }
 
-    $self->SUPER::BUILDARGS(@_);
+    $self->$orig(@_);
```
+1

karenetheridge commented on lib/Path/Dispatcher/Rule.pm:
```diff
@@ -12,17 +13,18 @@ has payload => (
 sub block { shift->payload(@_) }
-sub has_block { shift->has_payload(@_) }
-override BUILDARGS => sub {
+sub has_block { shift->has_payload(@_) };
```
`;` is unnecessary

karenetheridge commented on lib/Path/Dispatcher/Rule/Under.pm:
```diff
@@ -1,18 +1,18 @@
-    isa => 'Path::Dispatcher::PrefixRule',
+    isa => sub {
+        die "$_[0] not isa Path::Dispatcher::Rule"
+	    unless $_[0]->isa('Path::Dispatcher::Rule');
```
Type::Tiny::Class should be subtyped, rather than duplicating the `isa` check.

karenetheridge commented on lib/Path/Dispatcher/Rule/Under.pm:
```diff
@@ -1,18 +1,18 @@
+        die "$_[0] not isa Path::Dispatcher::Rule"
+	    unless $_[0]->isa('Path::Dispatcher::Rule');
+	
+        die "This rule ($_[0]) does not match just prefixes!"
```
The failure message should be pulled out as a separate attribute to the type declaration

karenetheridge commented on t/027-custom-named-captures.t:
```diff
@@ -5,18 +5,19 @@ use Path::Dispatcher;
     package My::Rule::NamedEnum;
-    use Any::Moose;
+    use Moo;
+    use Types::Standard qw( Str Int ArrayRef HashRef RegexpRef );
```
not all these imports are used

hashbangperl commented:
Thanks Karen, I'll work on it some more, most of the changes were from a previous PR that I reworked to fix a particular issue, I'll go over it again properly

Commit:
Updated isa constraints to use Type Utils class_type

Code review fixes

Commit:
Refactored isa constraints to use new inline type

Added new Type::Tiny type using parent constraint with Type::Utils class_type
Updated related test to match more specific error message
Based on code review feedback

Commit:
Small code review cleanup changes

hashbangperl commented:
Code review changes implemented

karenetheridge commented:
thanks, merged.

karenetheridge commented:
The switch from Moose was incomplete, as discovered when testing on a machine where Moose was not installed. fixing.

karenetheridge commented:
released as version 1.07.

hashbangperl commented:
Thanks Karen and nice catch! Glad that's sorted :)
